ext/gd/gd.c

```diff
@@ -3687,13 +3687,25 @@ PHP_FUNCTION(imageaffine)
 		if ((zval_affine_elem = zend_hash_index_find(Z_ARRVAL_P(z_affine), i)) != NULL) {
 			switch (Z_TYPE_P(zval_affine_elem)) {
 				case IS_LONG:
-					affine[i]  = Z_LVAL_P(zval_affine_elem);
+					affine[i] = Z_LVAL_P(zval_affine_elem);
+					if (ZEND_LONG_EXCEEDS_INT(affine[i])) {
+						zend_argument_value_error(2, "element %i must be between %d and %d", i, INT_MIN, INT_MAX);
+						RETURN_THROWS();
+					}
 					break;
 				case IS_DOUBLE:
 					affine[i] = Z_DVAL_P(zval_affine_elem);
+					if (ZEND_LONG_EXCEEDS_INT(affine[i])) {
```

Comment:
I don't think this will work on 32bit platforms, since `ZEND_LONG_EXCEEDS_INT()` is a no-op there, but users can pass a matrix like `[9223372036854775807., 1, 1, 1, 1, 1]`. Same for the `IS_STRING` case below.

Reply:
you re right I always forget.